connector/routingconnector/router.go

```diff
@@ -72,8 +68,48 @@ func newRouter[C any](
 }
 
 type routingItem[C any] struct {
-	consumer  C
-	statement *ottl.Statement[ottlresource.TransformContext]
+	consumer         C
+	statementContext string
+
+	resourceStatement *ottl.Statement[ottlresource.TransformContext]
+	logStatement      *ottl.Statement[ottllog.TransformContext]
```

Comment:
I think we could consider parameterizing this as per my previous comment.

Reply:
It might make sense here, since each route only needs one statement. Maybe something we can worry about later though?